benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/scheduler/TestTaskScheduler.java

```diff
@@ -181,17 +191,19 @@ public synchronized void handleRunningTest(String testID) {
         testRunningState = testModelDAO.getTestRunningState(testID);
         logger.info("handleRunningTest: prevTestState == " + prevTestState);
         logger.info("handleRunningTest: testState == " + testState);
+        logger.info("handleRunningTest: testRunningState == " + testRunningState);
+        logger.info("handleRunningTest: prevTestRunningState == " + prevTestRunningState);
 
         // Exit as soon as final state is executed
-        if (prevTestState != null && testState != null
-            && prevTestState.name().equals(testState.name())
-            && (testState == WAITING || testState == TERMINATED)) {
+        if (prevTestState == testState && (testState == WAITING || testState == TERMINATED)) {
           exit = true;
         }
         // Exit while waiting for the Experiment Manager to notify about the scheduled
         // experiment to be executed. This exits before the execution of HANDLE_EXPERIMENT_RESULT
-        else if (testRunningState != null
-            && testRunningState.equals(TestRunningState.HANDLE_EXPERIMENT_RESULT)) {
+        else if (testRunningState == TestRunningState.HANDLE_EXPERIMENT_RESULT
+            || testRunningState == TestRunningState.TERMINATING
+                && (prevTestRunningState == TestRunningState.HANDLE_EXPERIMENT_RESULT
```

Comment:
It is because in the case of `HANDLE_EXPERIMENT_RESULT` we should not execute the `TERMINATING` handler but wait for the trial to finish on Faban